### PR TITLE
Deprecate react-dates

### LINKED ISSUE #414: Deprecate react-dates
Now that we use ant design, which supports [DatePicker](https://ant.design/components/date-picker/), we should deprecate `react-dates`.

Components that need to be refactored or have code removed: 
- `app.jsx`
- `nav.jsx`
- `exportManager.jsx`
- `SingleDatePickerWrapper.jsx`
- `dashboardFilter_old.jsx` (are we still using this @enjeyw ??)
- `filter.jsx`

### PR DESCRIPTION
**Pull request by tristanhcole**
**Describe the Pull Request**
Deprecates `react-dates` in favour of ant design native datepicker component and closes #414.

Also removes the payment cycle from the frontend as we haven't used that component in 2 years and aren't using it across any current programs.
 
**Todo**

- [x] Link relevant [trello card](https://trello.com/b/PA2LhlOh/sempo-dev) or [related issue](https://github.com/teamsempo/SempoBlockchain/issues) to the pull request
- [x] Request review from relevant @person or team
- [x] QA your pull request. This includes running the app, login, testing changes etc.
- [x] Bump [backend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/config.py#L7) and/or [frontend](https://github.com/teamsempo/SempoBlockchain/blob/e3eb0f480e86d5a8ef2c1814127b70ff018671c4/app/package.json#L3) versions following Semver

### CODE REVIEW

File: app/client/components/transferAccount/transferAccountManager.jsx [hunk 1]
```diff
@@ -317,55 +314,6 @@ class TransferAccountManager extends React.Component {
                   <ManagerText>{this.state.one_time_code}</ManagerText>
                 </SubRow>
               </Row>
-              <Row style={{ margin: "0em 1em" }}>
```
Comment (tristanhcole) [inline app/client/components/transferAccount/transferAccountManager.jsx:320]: we haven't used this in two years. makes sense to remove from frontend and reimplement using ant-design if needed in future.

Comment (michielderoos) [reply app/client/components/transferAccount/transferAccountManager.jsx:320]: 🐜

**Review by michielderoos — APPROVED** [review]
QA'd, LGTM! 💯

**Comment by tristanhcole** [comment]
[Deprecate react-dates](https://trello.com/c/B98Cs5EM/679-deprecate-react-dates)

**Pull request merged**